elm-src/index.elm

```diff
@@ -0,0 +1,489 @@
+            String.concat
+                [ "<xml xmlns:xlink=\"http://www.w3.org/1999/xlink\"><para>"
+                , input
+                , "</para></xml>"
+                ]
+    in
+        case ExEmElm.Parser.parse text of
+            Ok doc ->
+                ExEmElm.Traverse.innerText (ExEmElm.Parser.root doc)
+
+            otherwise ->
+                ":)"
+
+
+describe_option : Option -> Html Msg
+describe_option option =
+    div [ class "search-details" ]
+        [ table []
+            [ tbody []
+                [ tr []
+                    [ th [] [ text "Description:" ]
+                    , td [] []
+                    , td [ class "description docbook" ] [ text (description_to_text option.description) ]
+                    ]
+                , tr []
+                    [ th [] [ text "Default Value:" ]
+                    , td [] []
+                    , td [ class "default" ] [ null_is_not_given nix option.default ]
+                    ]
+                , tr []
+                    [ th [] [ text "Example value:" ]
+                    , td [] []
+                    , td [ class "example" ] [ null_is_not_given nix option.example ]
+                    ]
+                , tr []
+                    [ th [] [ text "Declared In:" ]
+                    , td [] []
+                    , td [ class "declared-in" ] (List.map declaration_link option.declarations)
+                    ]
+                ]
+            ]
+        ]
+
+
+optionToTd : Maybe String -> ( String, Option ) -> List (Html Msg)
+optionToTd selected ( name, option ) =
+    let
+        isSelected =
+            (case selected of
+                Nothing ->
+                    False
+
+                Just selopt ->
+                    selopt == name
+            )
+    in
+        (List.concat
+            [ [ (tr []
+                    [ td
+                        [ onClick
+                            (if isSelected then
+                                (DeselectOption)
+                             else
+                                (SelectOption name)
+                            )
+                        ]
+                        [ tt [] [ text name ] ]
+                    ]
+                )
+              ]
+            , (if not isSelected then
+                []
+               else
+                [ tr []
+                    [ td [ class "details" ] [ describe_option option ]
+                    ]
+                ]
+              )
+            ]
+        )
+
+
+opt_tuple_matches_name : String -> ( String, Option ) -> Bool
+opt_tuple_matches_name term ( name, option ) =
+    term == name
+
+
+changePageIf : Bool -> Int -> String -> Html Msg
+changePageIf cond page txt =
+    button
+        (if cond then
+            [ onClick (ChangePage page) ]
+         else
+            [ disabled True ]
+        )
+        [ text txt ]
+
+
+view : Model -> Html Msg
+view model =
+    if List.length model.options > 0 then
```

Comment:
of `options` could get very long, `if not <| List.isEmpty model.options then` may be clearer/safer. `length` is `O(n)`

Reply:
`options` is about 5,000 items and will probably only go up (but slowly.)

Reply:
I think I'd either write this using `if List.isEmpty model.options then`, or using a `case..of`

```elm
case model.options of
    [] ->
        do stuff

    _ ->
        do other stuff
```